Mathlib/Topology/Instances/AddCircle/Defs.lean

```diff
@@ -534,14 +528,29 @@ theorem card_addOrderOf_eq_totient {n : ℕ} :
       n.totient_eq_card_lt_and_coprime]
     simp only [Nat.gcd_comm]
 
+omit [Fact (0 < p)]
+
+theorem finite_torsion {n : ℕ} (hn : 0 < n) : { u : AddCircle p | n • u = 0 }.Finite := by
+  wlog hp : 0 < p generalizing p
+  on_goal 2 =>
+    have := Fact.mk hp
```

Comment:
Here's how I'd write this. This is subjective, so feel free to accept or reject this as you choose.
```suggestion
  wlog hp : 0 < p generalizing p; swap
  · have := Fact.mk hp
```

Reply:
Thanks, I like this too!